core/state_processor.go

```diff
@@ -74,13 +75,17 @@ func (p *StateProcessor) Process(block *types.Block, parent *types.Header, state
 		blockNumber = block.Number()
 		allLogs     []*types.Log
 		gp          = new(GasPool).AddGas(block.GasLimit())
+		timestamp   = new(big.Int).SetUint64(header.Time)
 	)
 
+	// Configure the allow list if it goes into effect during the transition from [parent] to [block].
+	precompile.CheckConfigure(new(big.Int).SetUint64(parent.Time), timestamp, p.config.AllowListConfig, statedb)
```

Comment:
nit: is there some config array to iterate over here so the addition of more precompiles don't require changing code structure?

Reply:
Not yet, this is exactly what I was thinking of: adding a function to ChainConfig that returns a slice of these that can be easily iterated over.